src/buildHTML.js

```diff
@@ -151,27 +153,31 @@ const traverseNonSpaceNodes = function(
         const partialGroup = checkPartialGroup(node);
         if (partialGroup) { // Recursive DFS
             // $FlowFixMe: make nodes a $ReadOnlyArray by returning a new array
-            traverseNonSpaceNodes(partialGroup.children, callback, prev);
+            traverseNonSpaceNodes(partialGroup.children,
+                callback, prev, null, isRoot);
             continue;
         }
 
         // Ignore explicit spaces (e.g., \;, \,) when determining what implicit
         // spacing should go between atoms of different classes
-        if (node.classes[0] === "mspace") {
-            continue;
-        }
-
```

Comment:
Might I suggest that we keep the `continue statement` and change line 160 to:
~~~
if (node.classes[0] === "mspace" && !(isRoot && node.hasClass("newline")))
~~~

The current PR adds a superfluous `<span class="mspace newline"></span>` element after bins. It doesn't change the rendering, but also doesn't need to be there.

By retaining the `continue`, some of the `if (nonspace)` logic can be deleted.

Reply:
@ronkok Could you elaborate on the problem here? I've been unable to reproduce it, trying both `x+\\y` and `x+y`.  In the former, the `<span class="mspace newline"></span>` is needed to force the line break, and the rendering hasn't changed since the last release.

Reply:
Sorry. I should have been more clear. Given the input: `{d +\\ e}`, this PR inserts two space elements after the "+" sign, one with classes  `mspace newline`, and a second one with the desired spacing. The first element is redundant.

As currently written, this PR will always execute lines 181 - 184. By keeping the `continue` statement, that execution could be evaded and no redundant element inserted.

Reply:
I tried reinserting the `continue`, and it didn't make any difference. Indeed, the issue you point out is in the current release too, so it's old. The only code that executes without the continue is a redefinition of `prev.insertAfter`, so I believe it just affects where the next nonspace token will get placed (if it has any affect at all -- I haven't seen one).

Reply:
Well, it's not important. KaTeX writes a lot of elements. I withdraw my suggestion.